### DISCUSSION TITLE
Why it only processes one request at a time?

### DISCUSSION THREAD

**Discussion by iamumairayub** [FlareSolverr/FlareSolverr#729] [Ideas]
FlareSolverr works perfectly but has one shortcoming, it doesn't have ability to process more than one request at same time.

Here is code I tested, I could see logs that it was just processing one request at one time, once response from one request arrived, only then it processed next request.

from concurrent import futures

```
def send_request():
    headers = {
        'Content-Type': 'application/json',
    }
    json_data = {
        'cmd': 'request.get',
        'url': 'https://www.google.com'
    }
    response = requests.post('http://localhost:8191/v1', headers=headers, json=json_data)
    print(response.text)

all_jobs=[]
with futures.ThreadPoolExecutor(max_workers=10) as executor:
    for i in range(0,10000):
        job = executor.submit(
            send_request, 
        )
        all_jobs.append(job)

    for future in futures.as_completed(all_jobs):
        print('run completed: {}'.format(future.result()))
```

I also tried https://github.com/FlareSolverr/FlareSolverr/pull/674 version that supports sessions, I created multiple sessions, and sent pages to each session, but only page gets processed at one time.

Any workaround to make it work like Scrapy so it can process multiple requests at same time?

**Comment by paladev** [comment]
I'm not a python developer, but I think it's about GIL. Try using multiprocessing instead of threads.
